### ISSUE TITLE
Google Play Internal Testing error, production hash, App Store Signing

### ISSUE THREAD

**Issue by ljunquera** [AzureAD/microsoft-authentication-library-for-android#1543]
I am using AAD B2C in an Android app, which is working on my dev machine (Macbook Pro) but is crashing when I deploy it on Google Play for Internal Testing. The error I am getting is:

`The  redirect URI in the configuration file doesn't match with the one generated with package name and signature hash. Please verify the uri in the config file and your app registration in the Azure portal.`

When I am calling:
`PublicClientApplication.createMultipleAccountPublicClientApplication`

Following [these directions](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app#configure-platform-settings) I have generated a development Signature Hash. There is another link to generating a production Signature Hash.

`keytool -exportcert -alias SIGNATURE_ALIAS -keystore PATH_TO_KEYSTORE | openssl sha1 -binary | openssl base64`

but I am using Play App Signing so what do I do in this case?

I assume the issue is related to the production signature hash. I also don't understand how you can have a dev and prod B2C tenant if the hash is in the AndroidManifest.xml and the raw configuration file (raw/auth_config_multiple_account.json).

**Comment by paulkagiri** [comment]
@ljunquera for the first question, since you are using Play App Signing, head over to Google Play Console -> Play App Signing and pick the SHA-1 certificate fingerprint and generate a hash from it using various ways such as `echo {SHA-1 FINGERPRINT} | xxd -r -p | openssl base64` and use that in your App.

For the second question, you can use build type specific res values to dynamically switch between dev and test/production hashes such as:
```
android {
	...
	buildTypes{
		release {
				...
				resValue 'string', 'hash_name', 'hash from the above step'
		}

		debug{
				...
				resValue 'string', 'hash_name', 'the development hash'
		}
	}
}
```

Reference these in your manifest using `@string/hash_name`

**Comment by ljunquera** [comment]
Thanks, @paulkagiri. I think I got that one answered. 

**Android Manifest**
In the build.gradle I added this for each buildType where needed:
`manifestPlaceholders = [signatureHash:"<your signature hash>"]`

and this in the AndroidMainfest.xml:
`android:path="${signatureHash}"`



**Configuration File**
[This article ](https://www.journaldev.com/21533/android-build-types-product-flavors)and [this documentation](https://developer.android.com/studio/build/build-variants#product-flavors) helped.

I went into the project view in Project explorer, created a debug directory (to match my debug buildType in the build.gradle) in the src directory (making it a peer of main, test and androidTest), created a res directory under the new debug directory and a raw directory under the new res directory. I copied the auth_config_multiple_account.json file into it and altered the configuration to test it. All worked ok.
